src/network.c

```diff
@@ -851,3 +851,24 @@ int setup_SSL_connection(Connection *conn, SSL_CTX *ctx) {
   } while (ssl_ret != 1);
   return 0;
 }
+
+/**
+ * Sets 5 minute read and write timeouts on the passed-in socket.  Calls exit()
+ * if setsockopt fails for any reason.
+ * @param sockfd the socket to modify
+ */
+void set_socket_timeout_or_die(int sockfd) {
+  struct timeval timeout;
+  timeout.tv_sec = 300;
```

Comment:
There are possible cases I can imagine where a process could take up to 30-60 seconds. There are no possible cases where it could take 5 minutes. So how much more than 30-60 seconds is right? Well, I don't know where the exact line is. But I felt (and feel) confident that 5 minutes is on the other side of that line. So I chose 5 minutes.